main/tests/test_views.py

```diff
@@ -0,0 +1,127 @@
+        response = self.client.get('/post/1/publish/', follow=True)
+        # The same situation as with new post test
+        self.assertEqual(response.redirect_chain[0][0], '/post/1/')
+
+    def test_delete_post(self):
+        """Testing deleting post before and after login."""
+        response = self.client.get('/post/1/remove/')
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/post/1/remove/')
+        self.assertEqual(404, response.status_code)
+        Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        response = self.client.get('/post/1/remove/', follow=True)
+        self.assertEqual('/', response.redirect_chain[0][0])
+
+    def test_add_comment(self):
+        """Add comment to post."""
+        self.post = Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        resqonse = self.client.get('/post/1/comment/')
+        self.assertEqual(200, resqonse.status_code)
+        response = self.client.post('/post/1/comment/', {'author': self.admin, 'text': 'Super',
+                                                         'created_date': timezone.now(), 'is_approved': False},
+                                    follow=True)
+        self.assertEqual('/post/1/', response.redirect_chain[0][0])
+
+    def test_comment_aprove(self):
+        """Testing comment approve view."""
+        self.post = Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        self.comment = Comment.objects.create(post=self.post, author=self.admin, text='superComment',
+                                              created_date=timezone.now(), is_approved=False)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/comment/1/approve/', follow=True)
+        self.assertEqual('/post/1/', response.redirect_chain[0][0])
+
+    def test_comment_delete(self):
+        """Testing delete comment view."""
+        self.post = Post.objects.create(author=self.admin, title='Test', text='superText', created_date=timezone.now())
+        self.comment = Comment.objects.create(post=self.post, author=self.admin, text='superComment',
+                                              created_date=timezone.now(), is_approved=False)
+        authorization = self.client.login(username='admin', password='12345678')
+        self.assertTrue(authorization)
+        response = self.client.get('/comment/1/remove/', follow=True)
+        self.assertEqual('/post/1/', response.redirect_chain[0][0])
+
+    def tearDown(self):
+        """Clean data after each test."""
+        del self.client
```

Comment:
As I understood from sites where I read info about cleaning db in djnago, it cleans automatically if we use django.test.TestCase